acceptancetests/assess_network_health.py

```diff
@@ -176,17 +184,214 @@ def connect_to_existing_model(self, client, target_model):
         if client.show_model().keys()[0] is not target_model:
             client.switch(target_model)
 
+    def assign_spaces(self, client):
+        """Assigns spaces to subnets
+        Currently this only supports EC2.
+        Name the spaces sequentially: space1, space2, space3, etc.
+        We require at least 3 spaces.
+
+        :param client: Juju client object with controller
+        """
+        if client.env.provider == "ec2":
+            log.info("Assigning network spaces on EC2")
+            subnets = yaml.safe_load(client.get_juju_output('list-subnets',
+                '--format=yaml'))
+            if not subnets:
+                raise Exception('No subnets defined in EC2')
+            idx = 0
+            for subnet in subnets['subnets'].keys():
+                # skip subnets with INFAN in the provider-id as they may be
+                # inherited from underlay and therefore we cannot assign a space
+                if 'INFAN' not in subnets['subnets'][subnet]['provider-id']:
```

Comment:
Maybe be more verbose here in our logging. Ensure we are logging if we skip or add a space and why.